### ISSUE TITLE
Clarify documentation: `new Date()` saved locally will become a timestamp in firestore

### ISSUE THREAD

**Issue by usu** [mesqueeb/vuex-easy-firestore#108]
Firebase timestamps are properly converted to JS Date when data is loaded from server  (as described in https://mesqueeb.github.io/vuex-easy-firestore/extra-features.html#firestore-timestamp-conversion).

However, data thas is locally added to vuex store is not converted. Is there any method to ensure getters always convert Firebase timestamp objects to JS Date objects?

(related to #46)

**Comment by mesqueeb** [comment]
Hi! Merry christmas 🎄 
I'll need to check the best way to solve this. Maybe i'll make a helper class that will save new Date() in the Vuex store but a timestamp in Firestore. 

Sorry I can't use the computer for another whole day. 

I wonder though, if you just set the field to new Date() doesn't Firestore change it into a timestamp automatically on the server side? 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by usu** [comment]
Hi and merry Christmas, too :)

Thanks for your feedback. I might have misunderstood how 'convertTimestamps' is supposed to work. Let me try to explain differently.

I use convertTimestamps as described in the documentation. Firebase is in timestamp mode.

```javascript
const $db = firebase.firestore();
$db.settings({
  timestampsInSnapshots: true,
});

module = {
  // vuex-easy-firestore config
  serverChange: {
    convertTimestamps: {
      dateField: '%convertTimestamp%',
      updated_at: '%convertTimestamp%',
      created_at: '%convertTimestamp%',
    },
  }
}
```

```javascript
dispatch('module/set', {id, dateField: new Date()})
```

An API insert call is triggered and the doucment is inserted into Vuex (mutation INSERT_DOC):
```javascript
{
    "id": "julu1bee4t4h0t4mGHdT",
    "dateField": "[native Date Mon Dec 31 2018 00:00:00 GMT+0100 (Mitteleuropäische Normalzeit)]"
}
```

Immediately after that, mutation PATCH_DOC is triggered. Vuex now looks like this:
```javascript
{
    "id": "julu1bee4t4h0t4mGHdT",
    "dateField": {
        "seconds": 1546210800,
        "nanoseconds": 0
    },
    "created_at": {
        "seconds": 1545812147,
        "nanoseconds": 526000000
    }
}
```

If I now reload the page, data is freshly fetched from the server and dates are properly converted from timestamps to Date objects.
```javascript
{
    "id": "julu1bee4t4h0t4mGHdT",
    "dateField": "[native Date Mon Dec 31 2018 00:00:00 GMT+0100 (Mitteleuropäische Normalzeit)]",
    "created_at": "[native Date Wed Dec 26 2018 09:15:47 GMT+0100 (Mitteleuropäische Normalzeit)]",
    "updated_at": null
}
```

Looks like conversion from (firebase) timestamps to (JS) dates is done while inserting but not while patching. This leads to inconsistent formats in Vuex and application logic somehow has to deal with it.

Of course I can disable convertTimestamps alltogether and have my applicatoin logic doing the conversion. But I wanted to avoid calling ``.toDate()` all over the application.

Anything I am doing wrong here?

**Comment by mesqueeb** [comment]
@usu I will double check. In my application it works as expected but not in yours. 

I want to understand exactly what happens so tell me: does the problem exist when doing 'module/patch' on an existing object as well? Or only when a document is inserted first? 

Just to be sure also do "npm i vuex-easy-firestore@latest"

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by mesqueeb** [comment]
@usu I found the bug and have updated the library.
Should be fixed in latest version.

Please try `npm i vuex-easy-firestore@latest`
and let me know!

If it is fixed, please close the issue! ♡

**Comment by usu** [comment]
@mesqueeb Awesome, works perfect now. Thanks a lot for this ultra-fast fix.

(Fixed with commit 0b484dede7189ca4482b0cd43a53b8a31ac216e8)
